### ISSUE TITLE
Improve documentation memory usage warning?

### ISSUE THREAD

**Issue by HadrienG2** [STEllAR-GROUP/hpx#2040]
As was pointed out [multiple](https://github.com/STEllAR-GROUP/hpx/issues/950) [times](https://github.com/STEllAR-GROUP/hpx/issues/497) before, compiling HPX has quite large memory requirements. Since improving these does not seem to be a priority at the moment, it is best to document them well.

Currently, the documentation has [the following](http://stellar-group.github.io/hpx/docs/html/hpx/manual/build_system/building_hpx/build_recipes.html) to say on the matter :

> It is advisable that approximately 2 GB of memory per parallel process is available.

I can attest that this is quite accurate for the "make all" target, either in debug or release mode. However, today, when attempting a "make tests" on a debug build, I reached an new record 3 GB per parallel process. Unsurprisingly, the build crashed.

I think the documentation could warn about this better, but I am not sure how. Maybe the requirement should be turned into a lower bound, rather than an absolute number, i.e.

```diff
-approximately 2 GB of memory per parallel process
+at least 2 GB of memory per parallel process
```

What do you think?

**Comment by hkaiser** [comment]
We'd be happy to accept a PR to change this.

**Comment by HadrienG2** [comment]
Here's a PR for my proposal above. Again, if someone has a better idea of how to express that requirements can vary a lot depending on what is being built and which compiler is used, I'm all ears.

**Comment by hkaiser** [comment]
This has been fixed by merging #2042
